core/src/main/java/tech/pegasys/eth2signer/core/signing/LoadedSigners.java

```
@@ -0,0 +1,111 @@
+
+    final AzureKeyVaultSignerFactory azureFactory = new AzureKeyVaultSignerFactory();
+    final HashicorpConnectionFactory hashicorpConnectionFactory =
+        new HashicorpConnectionFactory(vertx);
+
+    final BlsArtifactSignerFactory blsArtifactSignerFactory =
+        new BlsArtifactSignerFactory(
+            config.getKeyConfigPath(), metricsSystem, hashicorpConnectionFactory);
+
+    final Secp256k1ArtifactSignerFactory ethSecpArtifactSignerFactory =
+        new Secp256k1ArtifactSignerFactory(
+            hashicorpConnectionFactory,
+            config.getKeyConfigPath(),
+            azureFactory,
+            EthSecpArtifactSigner::new);
+
+    final Secp256k1ArtifactSignerFactory fcSecpArtifactSignerFactory =
+        new Secp256k1ArtifactSignerFactory(
+            hashicorpConnectionFactory,
+            config.getKeyConfigPath(),
+            azureFactory,
+            signer -> new FcSecpArtifactSigner(signer, config.getFilecoinNetwork()));
+
+    final Collection<ArtifactSigner> signers =
+        SignerLoader.load(
+            config.getKeyConfigPath(),
+            "yaml",
+            new YamlSignerParser(
+                List.of(
+                    blsArtifactSignerFactory,
+                    ethSecpArtifactSignerFactory,
+                    fcSecpArtifactSignerFactory)));
+
+    return new LoadedSigners(
+        signers
+            .parallelStream()
+            .collect(
+                Collectors.groupingBy(
+                    i -> {
+                      if (i instanceof BlsArtifactSigner) {
+                        return SignerTypes.BLS;
+                      } else if (i instanceof EthSecpArtifactSigner) {
+                        return SignerTypes.ETH_SECP;
+                      } else if (i instanceof FcSecpArtifactSigner) {
+                        return SignerTypes.FC_SECP;
+                      } else {
+                        return SignerTypes.FC_BLS;
```

ideally we should throw an IllegalStateException if we encountered an invalid (or potentially new signer type).

---

done